src/main/java/life/qbic/io/commandline/OpenBISPasswordParser.java

```diff
@@ -0,0 +1,40 @@
+package life.qbic.io.commandline;
+
+import java.io.Console;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+public class OpenBISPasswordParser {
+
+  private static final Logger LOG = LogManager.getLogger(OpenBISPasswordParser.class);
+
+  /**
+   * Retrieve the password from input stream
+   *
+   * @return The password
+   */
+  public static String readPasswordFromInputStream() {
+    char[] password;
+    Console console = System.console();
+    if (console == null) {
+      LOG.error(
+          "Could not get console instance!"
+              + " Please make sure that you're running this from a normal console, a console supplied by an IDE will not suffice!");
+
+      return "";
+    }
+    password = console.readPassword();
+
+    return new String(password);
+  }
```

Comment:
```suggestion
  public static String readPasswordFromInputStream(InputStream inputStream) {
    String password;
    
    BufferedReader inputReader = new BufferedReader(new InputStreamReader(inputStream));
    password = inputReader.readLine();
    inputReader.close();
    
    return password;
  }
```
This is not tested. You might want to try it locally first.